src-electron/generator/matter/app/zap-templates/templates/app/helper.js

```diff
@@ -469,7 +469,10 @@ function chip_friendly_endpoint_type_name(options) {
 }
 
 function asMEI(prefix, suffix) {
-  return cHelper.asHex((prefix << 16) + suffix, 8);
+  // Left-shift (and for that matter bitwise or) produces a _signed_ 32-bit
+  // number, which will probably be negative.  Force it to unsigned 32-bit using
+  // >>> 0.
+  return cHelper.asHex(((prefix << 16) | suffix) >>> 0, 8);
```

Comment:
Is this basically a version of the solution I suggested above re: padding? If so, then it may be a possible to use concatenation of those asHex rather than int32toHex.

Reply:
This is the same code as above, but using `cHelper.asHex` instead of `bin.int32ToHex`, mostly because that's what the existing code at the two callsites was doing.  I don't know why we have those two separate conversion utilities....